### PR TITLE
Feat: APIs Completion(#40314)

### PR DESCRIPTION
**Pull request by Ankush-Negi**

### CODE REVIEW

File: src/controllers/trainee/routes.ts [hunk 1]
```diff
@@ -3,14 +3,283 @@ import TraineeController from './Controller';
...
 
 const traineeRouter: Router = Router();
-traineeRouter
 
-.get('/', authMiddleWare('getUsers','read'), validationHandler(validation.get),TraineeController.getAll)
-.get('/:id', authMiddleWare('getUsers','read'), validationHandler(validation.get), TraineeController.getById)
-.post('/', authMiddleWare('getUsers','read'), validationHandler(validation.create), TraineeController.create)
-.put('/:id', authMiddleWare('getUsers','read'), validationHandler(validation.update), TraineeController.update)
-.delete('/:id', authMiddleWare('getUsers','read'), validationHandler(validation.delete), TraineeController.delete);
+traineeRouter.use('/api', swaggerUi.serve, swaggerUi.setup(swaggerOptions));
```
Comment (VisaaliSuccessive) [inline src/controllers/trainee/routes.ts]: There should be a single route for the swagger doc.
i.e for user and trainee docs.

File: src/controllers/user/routes.ts [hunk 2]
```diff
@@ -1,8 +1,95 @@
...
 import UserController from './Controller';
 import authMiddleWare from '../../libs/routes/authMiddleWare';
+import swaggerOptions from '../../libs/swagger';
+import * as swaggerUi from 'swagger-ui-express';
 
 const userRouter: Router = Router();
-userRouter
-.get('/me',authMiddleWare('getUsers','read'),UserController.me);
+
+userRouter.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerOptions));
```
Comment (VisaaliSuccessive) [inline src/controllers/user/routes.ts]: There should be a single route for the swagger doc.
i.e for user and trainee docs.

File: src/repositories/versionable/VersionableRepository.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+    async create(data) {
+        const id = await this.generateId();
+        const { password, ...rest } = data;
+        const hashPassword: string = await bcrypt.hash(password, 10);
+        return this.modelType.create({
+            originalId: id,
+            _id: id,
+            password: hashPassword,
+            createdBy: Date.now(),
```
Comment (VisaaliSuccessive) [inline src/repositories/versionable/VersionableRepository.ts]: createdBy should not be used to store the current date.

File: src/repositories/versionable/VersionableRepository.ts [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            throw new Error(`Record at id ${id} does not exist`);
+        }
+        await this.delete(id);
+        const { __v, ...rest } = record;
+        const newId = await this.generateId();
+        const d = { ...rest, ...dataToUpdate, _id: newId };
+        return await this.modelType.create({
+            ...d,
+            updatedAt: Date.now(),
+            updatedBy: Date.now(),
```
Comment (VisaaliSuccessive) [inline src/repositories/versionable/VersionableRepository.ts]: updatedBy should not be used to store the current date.

File: src/controllers/trainee/routes.ts (hunk 1 above)
Comment (Ankush-Negi) [reply src/controllers/trainee/routes.ts]: Done

**Review by VisaaliSuccessive — APPROVED** [review]

File: src/controllers/user/routes.ts (hunk 2 above)
Comment (Ankush-Negi) [reply src/controllers/user/routes.ts]: Done

File: src/repositories/versionable/VersionableRepository.ts (hunk 3 above)
Comment (Ankush-Negi) [reply src/repositories/versionable/VersionableRepository.ts]: Done

File: src/repositories/versionable/VersionableRepository.ts (hunk 4 above)
Comment (Ankush-Negi) [reply src/repositories/versionable/VersionableRepository.ts]: Done

**Pull request merged**